`JodaModule` SPI metadata removed in 3.0.0-rc6

## Problem
- The `META-INF/services/tools.jackson.databind.JacksonModule` was removed in [commit 7e95c07](https://github.com/FasterXML/jackson-datatype-joda/commit/7e95c07).

## Impact 
- `JsonMapper.builder().findAndAddModules()` no longer discovers `JodaModule`.

## Proposed Fix
- Restore the `META-INF/services/tools.jackson.databind.JacksonModule`

I ran into the problem of jackson not discovering the `JodaModule` via `ServiceLoader`, and looked into it a bit further.

It appears that:
* `ServiceLoader` only uses `module-info` if the jar is on the _module-path_. ([ref](https://docs.oracle.com/en/java/javase/25/docs/api//java.base/java/util/ServiceLoader.html#deploying-service-providers-as-modules-heading))
* If the jar is on the _classpath_, then `ServiceLoader` uses the appropriate `META-INF/services/...` file.  ([ref](https://docs.oracle.com/en/java/javase/25/docs/api//java.base/java/util/ServiceLoader.html#deploying-service-providers-on-the-class-path-heading))

Therefore, to ensure that jackson modules are discoverable whether they are on the module-path or the classpath, the jackson module jars need to provide both the `module-info` metadata and the `META-INF/services/tools.jackson.databind.JacksonModule` file.

Since jackson is often deployed on the classpath, I think it makes sense for jackson module jars to also provide the `META-INF/services/tools.jackson.databind.JacksonModule` file.

Here's a minimal project with a unit test that attempts to discover the `JodaModule`, but fails because it is on the classpath, not the module-path

[jackson-datatype-joda-test.zip](https://github.com/user-attachments/files/25727074/jackson-datatype-joda-test.zip)

to execute (using Java 25):

```
./mvnw test
```

output shows the error:

```
[ERROR]   JodaModuleDiscoveryTest.ensureJodaModuleIsDiscoverableViaServiceLoader:23
Expecting ArrayList:
  []
to contain:
  ["tools.jackson.datatype.joda.JodaModule"]
but could not find the following element(s):
  ["tools.jackson.datatype.joda.JodaModule"]
```

--- a/release-notes/CREDITS
+++ b/release-notes/CREDITS
@@ -4,4 +4,15 @@ Cersion numbers in brackets indicate release in which the problem was fixed.
 
 (note: for older credits, see `CREDITS-2.x` instead)
 
-Tatu Saloranta, tatu.saloranta@iki.fi: author
+Author: Tatu Saloranta, tatu.saloranta@iki.fi
+
+Co-Authors (with only partial listings below):
+
+* Joo Hyuk Kim (@JooHyukKim)
+
+----------------------------------------------------------------------------
+
+
+Jooyoung Pyoung (@anthologia)
+ * Reported #175: `JodaModule` SPI metadata removed in 3.0.0-rc6
+  [3.1.1]

--- a/release-notes/VERSION
+++ b/release-notes/VERSION
@@ -9,6 +9,11 @@ Major version: 3 (for earlier, see VERSION-2.x)
 
 No changes since 3.1
 
+3.1.1 (not yet released)
+
+#175: `JodaModule` SPI metadata removed in 3.0.0-rc6
+ (reported by Jooyoung P)
+
 3.1.0 (23-Feb-2026)
 
 No changes since 3.0

--- a/src/main/resources/META-INF/services/tools.jackson.databind.JacksonModule
+++ b/src/main/resources/META-INF/services/tools.jackson.databind.JacksonModule
@@ -1,0 +1,1 @@
+tools.jackson.datatype.joda.JodaModule

